index.bs

```diff
@@ -491,6 +500,49 @@ spec: SHA2; urlPrefix: http://csrc.nist.gov/publications/fips/fips180-4/fips-180
   to attempt to use it without CORS. Additionally, user agents SHOULD report a
   warning message to the developer console to explain this failure.
 
+  ## Inline Verification Algorithms ## {#inline-verification}
+
+  Integrity may be enforced on a given <{script}> or <{style}> element by comparing an element's
+  [=child text content=] with the integrity metadata asserted via its <{script/integrity}>
+  attribute, as follows:
+
+  <div algorithm="inline verification">
+      To <dfn local-lt="inline verification" abstract-op>determine whether an element's inline
+      behavior should be blocked by integrity checks</dfn>, given an {{Element}} (|el|) and a
+      [=string=] (|body|), perform the following steps, which return "`Blocked`" or "`Allowed`":
+
+      1.  Assert: |el| is an {{HTMLOrSVGScriptElement}}, or an {{HTMLStyleElement}}.
+
+      2.  Let |metadata| be the value of [$get strongest|getting the strongest metadata$] from
+          [$parse metadata|parsing$] |el|'s <{script/integrity}> attribute.
+
+      3.  If |metadata| is empty, then return "`Allowed`".
+
+      4.  [=list/iterate|For each=] |item| in |metadata|:
+
+          1.  Let |algorithm| be the |alg| component of
+              |item|.
+
+          2.  Let |expectedValue| be the |val| component of
+              |item|.
+
+          3.  Let |actualValue| be the result of [=base64 encoding=] the result of executing
+              |algorithm| on |body|.
+
+          4.  If |actualValue| is |expectedValue|, return "`Allowed`".
+
+      5.  Return "`Blocked`".
+  </div>
+
+  ISSUE: This needs to be wired up to HTML's [=prepare a script=] algorithm by inserting a call to
+  this algorithm after the CSP check in step 13.
+
+  ISSUE: This needs to be wired up to HTML's [=update a style block=] algorithm by inserting a call
+  to this algorithm after the CSP check in step 5.
```

Comment:
Well, you'll also need to define it as a valid attribute for the `style` element if this is actually a thing we want to encourage.

Reply:
You're right. I thought we'd done that already, but of course it's only valid for `<link>`.